src/components/Footer.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+export default class Footer extends React.Component {
+    buttons = [
+        {name: 'all', label: 'All'},
+        {name: 'active', label: 'Active'},
+        {name: 'done', label: 'Done'}
+    ];
+    render() {
+        const {onFilterChange} = this.props;
+        let index = 0;
+        let done = this.props.done;
+        let clearCompleted;
```

Comment:
If you want to use this kind of render logic it is better to define new method. Let's consider `renderClearCompleted()` and `renderButtons()`

Reply:
It's usually not a good practice to move render logic in the component methods but it still good for training.